DigitalLearningSolutions.Data.Tests/DataServices/SupervisorDelegateDataServiceTests.cs

```diff
@@ -0,0 +1,123 @@
+            var logger = A.Fake<ILogger<SupervisorDelegateDataService>>();
+            supervisorDelegateDataService = new SupervisorDelegateDataService(connection, logger);
+        }
+
+        [Test]
+        public void GetSupervisorDelegateRecordByInviteHash_returns_correct_record()
+        {
+            // Given
+            var expectedRecord = new SupervisorDelegate
+            {
+                ID = 8,
+                SupervisorAdminID = 1,
+                DelegateEmail = "kevin.whittaker@hee.nhs.uk",
+                CandidateID = 254480,
+                Added = DateTime.Parse("2021-06-28 16:40:35.507"),
+                NotificationSent = DateTime.Parse("2021-06-28 16:40:35.507"),
+                Removed = null,
+                SupervisorEmail = "kevin.whittaker@hee.nhs.uk",
+                AddedByDelegate = false,
+                CentreId = 101,
+            };
+
+            // When
+            var result =
+                supervisorDelegateDataService.GetSupervisorDelegateRecordByInviteHash(
+                    inviteHashForFirstSupervisorDelegateRecord
+                );
+
+            // Then
+            result.Should().BeEquivalentTo(expectedRecord);
+        }
+
+        [Test]
+        public void GetPendingSupervisorDelegateRecordsByEmailsAndCentre_returns_correct_records()
+        {
+            using var transaction = new TransactionScope();
+
+            // Given
+            const int centreId = 101;
+            const string delegateEmailForValidRecord = "primary@email.com";
+            const string delegateEmailForRecordWithNonNullCandidateId = "kevin.whittaker@hee.nhs.uk";
+            const string delegateEmailForRemovedRecord = "louis.theroux@gmail.com";
+            const string? nullDelegateEmail = null;
+            var currentTime = DateTime.UtcNow;
+
+            connection.Execute(
+                @"INSERT INTO SupervisorDelegates (SupervisorAdminID, DelegateEmail, CandidateID, Added,
+                                 NotificationSent, Removed, SupervisorEmail, AddedByDelegate, InviteHash)
+                    OUTPUT INSERTED.ID
+                    VALUES (1, @delegateEmailForValidRecord, null, @currentTime, @currentTime, null,
+                            'supervisor@email.com', 0, '72e44c4d-77bd-4bed-a254-7cc27ab32928')",
+                new { delegateEmailForValidRecord, currentTime }
+            );
+
+            // When
+            var result = supervisorDelegateDataService.GetPendingSupervisorDelegateRecordsByEmailsAndCentre(
+                centreId,
+                new List<string?>
+                {
+                    delegateEmailForValidRecord,
+                    delegateEmailForRecordWithNonNullCandidateId,
+                    delegateEmailForRemovedRecord,
+                    nullDelegateEmail,
+                }
+            ).ToList();
+
+            // Then
+            result.Count.Should().Be(1);

```

Comment:
Multiple assertions should be wrapped in `using (new AssertionScope())`

Reply:
done :)